contrib/win32/win32compat/signal.c

```diff
@@ -260,59 +260,43 @@ sw_process_pending_signals()
 int
 wait_for_any_event(HANDLE* events, int num_events, DWORD milli_seconds)
 {
-	HANDLE all_events[MAXIMUM_WAIT_OBJECTS];
-	DWORD num_all_events;
+	HANDLE all_events[MAX_CHILDREN];
 	DWORD live_children = children.num_children - children.num_zombies;
+	DWORD num_all_events = num_events + live_children;
 	errno_t r = 0;
 
-	num_all_events = num_events + live_children;
-
-	if (num_all_events > MAXIMUM_WAIT_OBJECTS) {
+	if (num_all_events > MAX_CHILDREN) {
```

Comment:
We'll need a different MACRO named MAXIMUM_WAIT_OBJECTS_ENHANCED that can accommodate typical MAX_CHILDREN and total count of other events that can be passed to the wait* call. 
You'll also need to update SELECT_EVENT_LIMIT (currently at 32) - see w32_select() implementation.

Typical scenario that would hit the SELECT_EVENT_LIMIT is a client forwarding a whole bunch of local ports (in which case it has to listen on each such port - each of which ends up as an additional event in the wait() call via select())

Recommend the following:
`MAX_CHILDREN 512`
`SELECT_EVENT_LIMIT 512`
`MAXIMUM_WAIT_OBJECTS_ENHANCED 1024`

Reply:
Addressed in latest commit.